src/javascript/app/pages/user/set_currency.js

```diff
@@ -45,18 +45,37 @@ const SetCurrency = (() => {
                     })
                     .setVisibility(1);
             } else if (popup_action) {
-                const currencies = /multi_account|set_currency/.test(popup_action) ?
-                    getAvailableCurrencies(landing_company, payout_currencies) :
-                    getCurrencyChangeOptions(landing_company);
+                const is_virtual = Client.get('is_virtual');
+                const crypto_account = Client.hasCurrencyType('crypto');
+                let currencies = [];
+                
+                if (/multi_account|set_currency/.test(popup_action)) {
+                    if (is_virtual || crypto_account) {
+                        currencies = getVirtualAvailableCurrencies(landing_company, all_fiat);
+                    } else {
+                        currencies = getAvailableCurrencies(landing_company, payout_currencies);
+                    }
+                } else if (/switch_cryptocurrency/.test(popup_action)) {
+                    currencies = getCurrentCryptoCurrencies(all_fiat);
+                } else {
+                    currencies = getCurrencyChangeOptions(landing_company);
+                }
+
                 $('#hide_new_account').setVisibility(0);
+
+                if (currencies.length === 0){
```

Comment:
You can just write this instead: `if(!currencies?.length){}`

Reply:
good option, thanks 🙏